storey/table.py

```diff
@@ -1,25 +1,43 @@
+from contextlib import suppress
+from typing import List
 import copy
+import asyncio
 import math
 from datetime import datetime
-from typing import List
+from .drivers import Driver
+from .dtypes import FieldAggregator, SlidingWindows, FixedWindows, FlowError, _termination_obj
 
 from .aggregation_utils import is_raw_aggregate, get_virtual_aggregation_func, get_implied_aggregates, get_all_raw_aggregates, \
     get_all_raw_aggregates_with_hidden
-from .drivers import Driver
-from .dtypes import FieldAggregator, SlidingWindows, FixedWindows
 from .utils import _split_path, get_hashed_key
 
 
+class FlushInterval:
+    def __init__(self, interval_secs=300):
+        self.interval_secs = interval_secs
+
+
+class FlushAfterEveryEvent(FlushInterval):
+    def __init__(self):
+        FlushInterval.__init__(self, 0)
+
+
+_default_flush_interval = FlushInterval()
+
+
 class Table:
     """Table object, represents a single table in a specific storage.
 
     :param table_path: Path to the table in the storage.
     :param storage: Storage driver
     :param partitioned_by_key: Whether that data is partitioned by the key or not, based on this indication storage drivers
      can optimize writes. Defaults to True.
+    :param flush_interval: Policy indication when to flush the cache. default is after 300 secs
```

Comment:
I was going to suggest
`How often the cache will be flushed, in seconds. Default is 300 (5 minutes).`
but I see that it's not an int anymore.

What about disabling it? How is the user supposed to know what to pass in now that the type is `FlushInterval`?